<?xml version='1.0'?>
<robot name="legged_robot" xmlns:xacro="http://ros.org/wiki/xacro">

<!-- Included URDF/XACRO Files -->
<xacro:include filename="$(find legged_robot)/urdf/materials.urdf.xacro" />
<xacro:include filename="$(find legged_robot)/urdf/leg.urdf.xacro" />

<!--configuration-->
<property name="base_mass" value="5.0" />
<property name="base_x_size" value="0.5" />
<property name="base_y_size" value="0.3" />
<property name="base_z_size" value="0.05" />

<!--macro-->
<macro name="box_inertia" params="m x y z">
<inertia  ixx="${m*(y*y+z*z)/12.}" ixy = "0." ixz = "0."
iyy="${m*(x*x+z*z)/12.}" iyz = "0."
izz="${m*(x*x+z*z)/12.}" />
</macro>

<link name="base_footprint"/>

<gazebo reference="base_footprint">
<turnGravityOff>false</turnGravityOff>
</gazebo>

  <!--body-->
  <link name="base_link">
    <inertial>
    <origin xyz="0 0 0"/>
    <mass value="${base_mass}" />
    <box_inertia  m="${base_mass}" x="${base_x_size}" y="${base_y_size}" z="${base_z_size}"/>
    </inertial>
    <visual>
    <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
	<box size="${base_x_size} ${base_y_size} ${base_z_size}"/>
      </geometry>
    <material name="Gray" />
   </visual>
   <collision>
    <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
	<box size="${base_x_size} ${base_y_size} ${base_z_size}"/>
      </geometry>
    </collision>
  </link>

<joint name="base_link_joint" type="fixed">
<origin xyz="0 0 0" rpy="0 0 0" />
<parent link="base_footprint"/>
<child link="base_link" />
</joint>

<!--legs macro-->
<leg fr="front" lr="left" transX="1" transY="1"/>
<leg fr="front" lr="right" transX="1" transY="-1"/>
<leg fr="rear" lr="left" transX="-1" transY="1"/>
<leg fr="rear" lr="right" transX="-1" transY="-1"/>
  
  <!-- =========gazebo reference============ -->
<gazebo reference="base_link">
<material>Gazebo/Gray</material>
<turnGravityOff>false</turnGravityOff>
</gazebo>

<gazebo>
  <plugin name="gazebo_ros_legged_robot" filename="libgazebo_ros_legged_robot.so" >
    <commandTopic>/legged_robot/cmd_ang</commandTopic>
    <hinge_lf1>hinge_left_front_1</hinge_lf1>
    <hinge_lf2>hinge_left_front_2</hinge_lf2>
    <hinge_lf3>hinge_left_front_3</hinge_lf3>
    <hinge_lr1>hinge_left_rear_1</hinge_lr1>
    <hinge_lr2>hinge_left_rear_2</hinge_lr2>
    <hinge_lr3>hinge_left_rear_3</hinge_lr3>
    <hinge_rf1>hinge_right_front_1</hinge_rf1>
    <hinge_rf2>hinge_right_front_2</hinge_rf2>
    <hinge_rf3>hinge_right_front_3</hinge_rf3>
    <hinge_rr1>hinge_right_rear_1</hinge_rr1>
    <hinge_rr2>hinge_right_rear_2</hinge_rr2>
    <hinge_rr3>hinge_right_rear_3</hinge_rr3>
    <gain>0.1</gain>
    <hingeTorque>100.0</hingeTorque>
    <publishHingeTF>false</publishHingeTF>
  </plugin>
  </gazebo>
</robot>
